packages/suite/src/views/wallet/coinmarket/buy/offers/index.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { AppState } from '@suite-types';
+import { WalletLayout } from '@wallet-components';
+import styled from 'styled-components';
+import { ComponentProps, Props } from '@wallet-types/coinmarketBuyOffers';
+import { CoinmarketBuyOffersContext, useOffers } from '@wallet-hooks/useCoinmarketBuyOffers';
+import Offers from './Offers';
+
+const mapStateToProps = (state: AppState): ComponentProps => ({
+    selectedAccount: state.wallet.selectedAccount,
+    device: state.suite.device,
+    quotes: state.wallet.coinmarket.buy.quotes,
+    alternativeQuotes: state.wallet.coinmarket.buy.alternativeQuotes,
+    quotesRequest: state.wallet.coinmarket.buy.quotesRequest,
+    isFromRedirect: state.wallet.coinmarket.buy.isFromRedirect,
+    addressVerified: state.wallet.coinmarket.buy.addressVerified,
+    providersInfo: state.wallet.coinmarket.buy.buyInfo?.providerInfos,
+});
+
+const Wrapper = styled.div`
+    display: flex;
+    width: 100%;
+    flex-direction: column;
+`;
+
+const OffersIndex = (props: Props) => {
+    const { selectedAccount } = props;
+    if (props.selectedAccount.status !== 'loaded') {
+        return <WalletLayout title="Coinmarket | buy" account={selectedAccount} />;
+    }
+
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    const coinmarketOffersValues = useOffers({ ...props, selectedAccount });
```

Comment:
Hooks should always be placed at the top level of functions.

Reply:
We are also not very happy about it but we are just following practice started in other parts of the suite code, where the hook is used only after the selected account is loaded. See for example https://github.com/trezor/trezor-suite/blob/2ab75ac21dd3aac83b636c628b63e39f93570904/packages/suite/src/views/wallet/send/index.tsx#L35-L51

Reply:
i'll never understand this approach: "i know it's not right and i don't like it but i will copy/paste it anyway instead of make it better"

fixed in send:
https://github.com/trezor/trezor-suite/pull/2498

Reply:
Things can slip through reviews and I agree with Szymon here. In general, as a rule of thumb, ignores/exceptions should be avoided as rules are here for a reason. In this specific case, the reasoning is well explained in the React documentation: https://reactjs.org/docs/hooks-rules.html#explanation

Reply:
In general, I absolutely agree with you, @keraf. Ignores/exceptions should be avoided and hook rules should be followed.
However, our main goal is to deliver business (not software) in time and reasonable quality and any shortcut (especially if done by the principal developer of the Suite) at this stage is fine by us.
Anyway, thanks to this review, the bad practice is removed from all parts of code :)